src/test/java/api/requests/ItemsLostRequiringActualCostsTests.java

```diff
@@ -0,0 +1,211 @@
+import api.support.builders.LostItemFeePolicyBuilder;
+import api.support.http.IndividualResource;
+import api.support.http.ItemResource;
+import api.support.http.ResourceClient;
+import api.support.matchers.ItemMatchers;
+import api.support.spring.SpringApiTest;
+import io.vertx.core.json.JsonObject;
+
+public class ItemsLostRequiringActualCostsTests extends SpringApiTest {
+
+  public ItemsLostRequiringActualCostsTests() {
+    super(true, true);
+  }
+
+  @Test
+  public void isEmptyWhenNoLostItems() {
+    createClosedItem();
+    createOpenItem();
+
+    List<JsonObject> items = ResourceClient.forItemsLostRequiringActualCosts().getAll();
+
+    assertTrue(items.isEmpty());
+  }
+
+  @Test
+  public void hasManyItemsLost() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.chargeFee().getId());
+
+    final ItemResource closedItem = createClosedItem();
+    final ItemResource openItem = createOpenItem();
+    final ItemResource declaredLostItem = createDeclaredLostItem();
+
+    useLostItemPolicy(ageToLostAndBilledAfterOneMinute().getId());
+
+    final ItemResource agedToLostItem = createAgedToLostItem();
+
+    ageToLostFixture.chargeFees();
+
+    Collection<JsonObject> items = ResourceClient.forItemsLostRequiringActualCosts().getAll();
+
+    assertThat(items.size(), is(2));
+    validateItemExistsInItems(items, declaredLostItem);
+    validateItemInItemsIsDeclaredLost(items, declaredLostItem);
+    validateItemExistsInItems(items, agedToLostItem);
+    validateItemInItemsIsAgedToLost(items, agedToLostItem);
+    validateItemNotExistsInItems(items, closedItem);
+    validateItemNotExistsInItems(items, openItem);
+  }
+
+  @Test
+  public void hasItemDeclaredLost() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.chargeFee().getId());
+
+    final ItemResource declaredLostItem = createDeclaredLostItem();
+
+    Collection<JsonObject> items = ResourceClient.forItemsLostRequiringActualCosts().getAll();
+
+    assertThat(items.size(), is(1));
+    validateItemExistsInItems(items, declaredLostItem);
+    validateItemInItemsIsDeclaredLost(items, declaredLostItem);
+  }
+
+  @Test
+  public void hasItemAgedToLost() {
+    useLostItemPolicy(ageToLostAndBilledAfterOneMinute().getId());
+
+    final ItemResource agedToLostItem = createAgedToLostItem();
+
+    ageToLostFixture.chargeFees();
+
+    Collection<JsonObject> items = ResourceClient.forItemsLostRequiringActualCosts().getAll();
+
+    assertThat(items.size(), is(1));
+    validateItemExistsInItems(items, agedToLostItem);
+    validateItemInItemsIsAgedToLost(items, agedToLostItem);
+  }
+
+  @Test
+  public void hasNoUnbilledItemAgedToLost() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.ageToLostAfterOneMinute().getId());
+
+    createAgedToLostItem();
+
+    Collection<JsonObject> items = ResourceClient.forItemsLostRequiringActualCosts().getAll();
+
+    assertThat(items.size(), is(0));
+  }
+
+  @Test
+  public void hasNoBilledWithActualCostItemAgedToLost() {
```

Comment:
What situation is this test describing, the situation where an item that has been aged to lost has not been billed yet?

Reply:
I might try to re-word it to make this more clear.

This is testing for the case where:

For Aged to lost items...
- Item status is 'Aged to lost' _(matches)_
- lostItemHasBeenBilled (from Loan Record) is "true" _(misses)_
- No fee/fine 'Action' record exists with Fee/fine type of 'Lost item fee (actual cost)' _(matches)_
- Loan is still open_(matches)_

I need to ensure that when lostItemHasBeenBilled is FALSE, then it properly does not get included in the list.

Reply:
Scratch that. I was thinking of `hasNoUnbilledItemAgedToLost()`. They are both similarly named, I can see the confusion.

The function `hasNoBilledWithActualCostItemAgedToLost` is to be renamed to something like `hasNoAgedToLostBilledItemWithActualCost`.
Therefore it should instead be this situation:

For Aged to lost items...
- Item status is 'Aged to lost' _(matches)_
- lostItemHasBeenBilled (from Loan Record) is "true" _(matches)_
- No fee/fine 'Action' record exists with Fee/fine type of 'Lost item fee (actual cost)' _(misses)_
- Loan is still open _(matches)_